paasta_tools/kubernetes_tools.py

```diff
@@ -610,7 +659,7 @@ def raw_selectors_to_requirements(
                 raise ValueError(
                     f"Unknown k8s node affinity operator: {config['operator']}"
                 )
-            requirements.append((label, config["operator"], values))
+            requirements.append((label, cast(str, config["operator"]), values))
```

Comment:
huh, surprised that mypy can't tell that a Literal string is also a str

Reply:
I had a really long slack thread in #python about this, I think I'll use the [suggestion](https://yelp.slack.com/archives/CA8G37W3Y/p1702924064196629?thread_ts=1702681402.140179&cid=CA8G37W3Y) to add a type annotation to the definition of `requirements` instead.

I think the `cast` here was just a complicated way of getting mypy to infer the correct type for `requirements` anyway.